lib/features/children/family_goal_tracker/repository/goal_tracker_repository.dart

```diff
@@ -13,14 +13,14 @@ class GoalTrackerRepositoryImpl with GoalTrackerRepository {
   final APIService _apiService;
 
   @override
-  Future<List<FamilyGoal>> fetchFamilyGoal() async {
+  Future<FamilyGoal> fetchFamilyGoal() async {
     final response = await _apiService.fetchFamilyGoal();
-
-    final List<FamilyGoal> goals = [];
-
-    for (final item in response) {
-      goals.add(FamilyGoal.fromMap(item as Map<String, dynamic>));
+    FamilyGoal goal;
```

Comment:
18 till 24 can be simplified:

`if(response.isEmpty)
return FamilyGoal.empty();

return FamilyGoal.fromMap(response);`